internal/monitor/instance/linux/processor.go

```diff
@@ -45,6 +47,30 @@ func baseName(name, separator string) string {
 	return name[lastseparator+1:]
 }
 
+// RemapData Remaps the contextstore data from an old format to the newer format.
+func (l *linuxProcessor) RemapData(data string, fixedData interface{}) error {
+	event := &events.EventInfo{}
+
+	if err := json.Unmarshal([]byte(data), event); err != nil {
+		return fmt.Errorf("Received error %s while remapping data", err)
+	}
+	//Convert the eventInfo data to new format
+	for index, s := range event.Services {
+		if s.Port != 0 {
+			s.Ports = &portspec.PortSpec{
+				Min: s.Port,
+				Max: s.Port,
+			}
+		}
+		event.Services[index].Ports = s.Ports
+	}
+	if fixedData.(*StoredContext).Tags == nil {
```

Comment:
1) Potential panic in conversion
2) fixedData.(*StoredContext) should be converted once and temp var should be used after that.